reactioncommerce/reaction
# (fix:) Issue #3797 — Removing search-mongo plugin causes errors at startup

Resolves #3797   
Impact: **minor**  
Type: **bugfix**

## Issue
When replacing included plugin `reaction-search` with a custom one, Reaction cannot start due to errors about `reaction-search` being missing. These errors originate from the `reaction-migrations` plugin which tries to load `reaction-search` from its default path (`/imports/plugins/includes/search-mongo`), regardless of whether it exists or not.

## Solution
Lazy-load `reaction-search` if it is registered in the `Packages` collection using an `async` function, then run the standard migration scripts while checking that imported `reaction-search` functions exist before calling them.

I'm not sure if the code for this solution is optimal. I've considered returning the imported functions from `loadSearchRecordBuilderIfItExists` in an object, thus being passed to `then`'s callback as a parameter. Would that be a cleaner way of doing it?

Any input on this would be welcome.

## Breaking changes
None.


## Testing
1. Run `reaction reset -n && reaction`, Reaction Commerce should start fine
2. Rename `/imports/plugins/included/search-mongo` to `/imports/plugins/included/.search-mongo`
3. Run `reaction reset -n && reaction`, Reaction Commerce should start fine and output warnings indicating that search record building at step 1 and 14 of the migration scripts were skipped
